backend/src/middleware/softDeleteMiddleware.spec.js

```diff
@@ -1,82 +1,127 @@
-    instance.create('Category', {
-      id: 'cat9',
-      name: 'Democracy & Politics',
-      icon: 'university',
     }),
   ])
 
-  await factory.authenticateAs({ email: 'user@example.org', password: '1234' })
+  user = users[0]
+  moderator = users[1]
+  troll = users[2]
+
+  await neode.create('Category', {
+    id: 'cat9',
+    name: 'Democracy & Politics',
+    icon: 'university',
+  })
+
   await Promise.all([
-    factory.follow({ id: 'u2', type: 'User' }),
-    factory.create('Post', { id: 'p1', title: 'Deleted post', deleted: true, categoryIds }),
+    user.relateTo(troll, 'following'),
+    factory.create('Post', {
+      author: user,
+      id: 'p1',
+      title: 'Deleted post',
+      slug: 'deleted-post',
+      deleted: true,
+      categoryIds,
+    }),
     factory.create('Post', {
+      author: user,
       id: 'p3',
       title: 'Publicly visible post',
+      slug: 'publicly-visible-post',
       deleted: false,
       categoryIds,
     }),
   ])
 
   await Promise.all([
     factory.create('Comment', {
+      author: user,
       id: 'c2',
       postId: 'p3',
       content: 'Enabled comment on public post',
     }),
   ])
 
-  await Promise.all([factory.relate('Comment', 'Author', { from: 'u1', to: 'c2' })])
-
-  const asTroll = Factory()
-  await asTroll.authenticateAs({ email: 'troll@example.org', password: '1234' })
-  await asTroll.create('Post', {
+  await factory.create('Post', {
     id: 'p2',
+    author: troll,
     title: 'Disabled post',
     content: 'This is an offensive post content',
+    contentExcerpt: 'This is an offensive post content',
     image: '/some/offensive/image.jpg',
     deleted: false,
     categoryIds,
   })
-  await asTroll.create('Comment', { id: 'c1', postId: 'p3', content: 'Disabled comment' })
-  await Promise.all([asTroll.relate('Comment', 'Author', { from: 'u2', to: 'c1' })])
-
-  const asModerator = Factory()
-  await asModerator.authenticateAs({ email: 'moderator@example.org', password: '1234' })
-  await asModerator.mutate('mutation { disable( id: "p2") }')
-  await asModerator.mutate('mutation { disable( id: "c1") }')
-  await asModerator.mutate('mutation { disable( id: "u2") }')
+  await factory.create('Comment', {
+    id: 'c1',
+    author: troll,
+    postId: 'p3',
+    content: 'Disabled comment',
+    contentExcerpt: 'Disabled comment',
+  })
+
+  const { server } = createServer({
+    context: () => {
+      return {
+        driver,
+        neode,
+        user: authenticatedUser,
+      }
+    },
+  })
+  const client = createTestClient(server)
+  query = client.query
+  mutate = client.mutate
+
+  authenticatedUser = await moderator.toJson()
+  const disableMutation = gql`
+    mutation($id: ID!) {
+      disable(id: $id)
+    }
+  `
+  await Promise.all([
+    mutate({ mutation: disableMutation, variables: { id: 'c1' } }),
+    mutate({ mutation: disableMutation, variables: { id: 'u2' } }),
+    mutate({ mutation: disableMutation, variables: { id: 'p2' } }),
```

Comment:
> [<img alt="alina-beck" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/alina-beck) **Authored by [alina-beck](https://github.com/alina-beck)**
_<time datetime="2019-09-03T15:13:51Z" title="Tuesday, September 3rd 2019, 5:13:51 pm +02:00">Sep 3, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -1,82 +1,124 @@
-import { GraphQLClient } from 'graphql-request'
 import Factory from '../seed/factories'
-import { host, login } from '../jest/helpers'
-import { neode } from '../bootstrap/neo4j'
+import { gql } from '../jest/helpers'
+import { neode as getNeode, getDriver } from '../bootstrap/neo4j'
+import createServer from '../server'
+import { createTestClient } from 'apollo-server-testing'
 
 const factory = Factory()
-const instance = neode()
+const neode = getNeode()
+const driver = getDriver()
 
-let client
 let query
+let mutate
+let graphqlQuery
 let action
 const categoryIds = ['cat9']
+let authenticatedUser
+let user
+let moderator
+let troll
 
 beforeAll(async () => {
   // For performance reasons we do this only once
-  await Promise.all([
-    factory.create('User', { id: 'u1', role: 'user', email: 'user@example.org', password: '1234' }),
+  const users = await Promise.all([
+    factory.create('User', { id: 'u1', role: 'user' }),
     factory.create('User', {
       id: 'm1',
       role: 'moderator',
-      email: 'moderator@example.org',
       password: '1234',
     }),
     factory.create('User', {
       id: 'u2',
       role: 'user',
       name: 'Offensive Name',
+      slug: 'offensive-name',
       avatar: '/some/offensive/avatar.jpg',
       about: 'This self description is very offensive',
-      email: 'troll@example.org',
-      password: '1234',
-    }),
-    instance.create('Category', {
-      id: 'cat9',
-      name: 'Democracy & Politics',
-      icon: 'university',
     }),
   ])
 
-  await factory.authenticateAs({ email: 'user@example.org', password: '1234' })
+  user = users[0]
+  moderator = users[1]
+  troll = users[2]
+
+  await neode.create('Category', {
+    id: 'cat9',
+    name: 'Democracy & Politics',
+    icon: 'university',
+  })
+
   await Promise.all([
-    factory.follow({ id: 'u2', type: 'User' }),
-    factory.create('Post', { id: 'p1', title: 'Deleted post', deleted: true, categoryIds }),
+    user.relateTo(troll, 'following'),
+    factory.create('Post', {
+      author: user,
+      id: 'p1',
+      title: 'Deleted post',
+      slug: 'deleted-post',
+      deleted: true,
+      categoryIds,
+    }),
     factory.create('Post', {
+      author: user,
       id: 'p3',
       title: 'Publicly visible post',
+      slug: 'publicly-visible-post',
       deleted: false,
       categoryIds,
     }),
   ])
 
   await Promise.all([
     factory.create('Comment', {
+      author: user,
       id: 'c2',
       postId: 'p3',
       content: 'Enabled comment on public post',
     }),
   ])
 
-  await Promise.all([factory.relate('Comment', 'Author', { from: 'u1', to: 'c2' })])
-
-  const asTroll = Factory()
-  await asTroll.authenticateAs({ email: 'troll@example.org', password: '1234' })
-  await asTroll.create('Post', {
+  await factory.create('Post', {
     id: 'p2',
+    author: troll,
     title: 'Disabled post',
     content: 'This is an offensive post content',
+    contentExcerpt: 'This is an offensive post content',
     image: '/some/offensive/image.jpg',
     deleted: false,
     categoryIds,
   })
-  await asTroll.create('Comment', { id: 'c1', postId: 'p3', content: 'Disabled comment' })
-  await Promise.all([asTroll.relate('Comment', 'Author', { from: 'u2', to: 'c1' })])
-
-  const asModerator = Factory()
-  await asModerator.authenticateAs({ email: 'moderator@example.org', password: '1234' })
-  await asModerator.mutate('mutation { disable( id: "p2") }')
-  await asModerator.mutate('mutation { disable( id: "c1") }')
-  await asModerator.mutate('mutation { disable( id: "u2") }')
+  await factory.create('Comment', {
+    id: 'c1',
+    author: troll,
+    postId: 'p3',
+    content: 'Disabled comment',
+    contentExcerpt: 'Disabled comment',
+  })
+
+  const { server } = createServer({
+    context: () => {
+      return {
+        driver,
+        neode,
+        user: authenticatedUser,
+      }
+    },
+  })
+  const client = createTestClient(server)
+  query = client.query
+  mutate = client.mutate
+
+  authenticatedUser = await moderator.toJson()
+  const disableMutation = gql`
+    mutation($id: ID!) {
+      disable(id: $id)
+    }
+  `
+  await Promise.all([
+    mutate({ mutation: disableMutation, variables: { id: 'c1' } }),
+    mutate({ mutation: disableMutation, variables: { id: 'u2' } }),
+    mutate({ mutation: disableMutation, variables: { id: 'p2' } }),
```


I think @mattwr18 already mentioned this somewhere else but could you use more descriptive `ids` here as well?

And I'm guessing you can't disable posts on creating them (like you did with `deleted: true` above) or otherwise you would have done it, right? :wink:

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-09-03T16:32:52Z" title="Tuesday, September 3rd 2019, 6:32:52 pm +02:00">Sep 3, 2019</time>_

---


![Screenshot - 2019-09-03T182930 203](https://user-images.githubusercontent.com/2110676/64191397-b172d100-ce78-11e9-8b2f-7b23a34fceed.png)

It *should* not be possible. However, we use the augmented schema of [neo4j-graphql-js](https://github.com/neo4j-graphql/neo4j-graphql-js/) that adds the `CreatePost` mutation to our graphql schema. Silly enough, the mutation allows `deleted`, `disabled`, `createdAt`, `updatedAt` etc. as valid input params.

I will declare the `CreatePost` and similar mutations in the schema explicitly.  Any mutations/queries that are there already won't be augmented by the library.